hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -426,6 +428,16 @@ protected Pair<HoodieRecord, HoodieRecord> getFinalMergeRecords(BufferedRecord<T
       HoodieRecord newHoodieRecord = constructHoodieAvroRecord(recordContext, newerRecord, payloadClass);
       return Pair.of(oldHoodieRecord, newHoodieRecord);
     }
+
+    @Override
+    protected Option<Pair<HoodieRecord, Schema>> getMergedRecord(BufferedRecord<T> olderRecord, BufferedRecord<T> newerRecord, boolean isFinalMerge) throws IOException {
```

Comment:
did we add UTs for this at BufferedRecordMerger layer. essentially for the ExpressionPayloadRecordMerger

Reply:
I'm not sure how to add UTs for the expression payload. Currently it looks like all the existing testing is done in functional tests.